073-extend-feature-gates-to-all-operators.md

```diff
@@ -0,0 +1,114 @@
+# Extend Feature Gates to all Strimzi operators
+
+Strimzi introduced _Feature Gates_ in [Proposal no. 22](https://github.com/strimzi/proposals/blob/main/022-feature-gates.md).
+Feature gates allow to enable or disable some features or change the behavior of Strimzi.
+They also have a maturity model that allows the feature gates to progress as the features they introduce mature.
+For more details, please check out the [original proposal](https://github.com/strimzi/proposals/blob/main/022-feature-gates.md).
+The original proposal introduced feature gates initially only to the Strimzi Cluster Operator.
+This proposal provides the plan for extending them to all Strimzi operators.
+
+## Current situation
+
+Currently, the feature gates are used only in the Cluster Operator.
+The feature gates are configured using `STRIMZI_FEATURE_GATES` environment variable where the individual gates can be enabled or disabled.
+The configured feature gates are used across the Cluster Operator code.
+Also the code implementing them is part of the `cluster-operator` module.
+The feature gates do not extent to the other operators (User and Topic Operators).
+
+## Motivation
+
+Although the feature gates currently don't apply to the other operators, some of them still affect them:
+* For example, the `UnidirectionalTopicOperator` feature gate controlled the mode in which the Topic Operator was running.
+  But it was using different environment variables to configure the right mode Topic Operator and not the feature gates configuration.
+* In the future, the `UseServerSideApply` feature gate (see [Proposal no. 52](https://github.com/strimzi/proposals/blob/main/052-k8s-server-side-apply.md) for more details) will apply to at least the User Operator.
+
+Additional feature gates that apply to the other operators might be added in the future.
+To have a unified way how to manage such feature gates, this proposal provides the plan for extending the feature gates to the User and Topic Operators as well.
+
+## Proposal
+
+The feature gates and the `STRIMZI_FEATURE_GATES` environment variable will keep its current format.
+The support for the `STRIMZI_FEATURE_GATES` environment variable will be added to the User and Topic Operators.
+The same feature gates will be defined and handled by every operator.
+However, if the operator doesn't have any use for the feature gate, it can simply ignore it in its code.
+
+When the User and Topic Operators are deployed through the Cluster Operator as part of the Kafka cluster, the user will configure the feature gates as today using the `STRIMZI_FEATURE_GATES` environment variable.
+And the Cluster Operator will then automatically pass the environment variable to the User and Topic Operators when deploying them.
+When the User and Topic Operators are deployed as standalone, user can configure the `STRIMZI_FEATURE_GATES` environment variable directly in their deployments.
```

Comment:
To be clear, typically all the feature gate configuration is done in the Cluster Operator config. If standalone UO and TO are used, you configure in Cluster Operator and in the standalone operators?

Reply:
When using standalone User and/or Topic operators, they are completely independent and it is not really expected that you use the Cluster Operator at all. I tried to make it more clear in the proposl.

Reply:
Right. Makes sense now. Thanks